components/script/dom/domrectlist.rs

```diff
@@ -19,17 +18,17 @@ pub struct DOMRectList {
 }
 
 impl DOMRectList {
-    fn new_inherited(window: JSRef<Window>,
-                     rects: &RootedVec<JS<DOMRect>>) -> DOMRectList {
+    fn new_inherited<T>(window: JSRef<Window>, rects: T) -> DOMRectList
+                        where T: Iterator<Item=Temporary<DOMRect>> {
         DOMRectList {
             reflector_: Reflector::new(),
-            rects: (**rects).clone(),
+            rects: rects.map(JS::from_rooted).collect(),
             window: JS::from_rooted(window),
         }
     }
 
-    pub fn new(window: JSRef<Window>,
-               rects: &RootedVec<JS<DOMRect>>) -> Temporary<DOMRectList> {
+    pub fn new<T>(window: JSRef<Window>, rects: T) -> Temporary<DOMRectList>
```

Comment:
I'm not sure if we should be using generics for `new`/`new_inherited`. I don't mind much, but there might be some style we're trying to maintain. @jdm @Ms2ger ?